packages/jest-haste-map/src/crawlers/node.js

```diff
@@ -0,0 +1,83 @@
+ /**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+'use strict';
+
+const denodeify = require('denodeify');
+const fs = require('graceful-fs');
+const path = require('../fastpath');
+
+const statFile = denodeify(fs.stat);
+const readDir = denodeify(fs.readdir);
+
+function nodeCrawl(roots, extensions, ignorePattern, data) {
+  const extensionsPattern = new RegExp('\.(' + extensions.join('|') + ')$');
+  const files = Object.create(null);
+  const queue = roots.slice();
```

Comment:
In Jest at least, we have `testPathDirs` and each of them is a root. On www it is more practical to crawl certain folders rather than all of them.